tests/baselines/reference/genericRestParameters3.errors.txt

```diff
@@ -69,16 +70,17 @@ tests/cases/conformance/types/rest/genericRestParameters3.ts(59,5): error TS2345
     ~~
 !!! error TS2322: Type '(x: string, y: string) => void' is not assignable to type '(x: string, ...args: [string] | [number, boolean]) => void'.
 !!! error TS2322:   Types of parameters 'y' and 'args' are incompatible.
-!!! error TS2322:     Type '[string] | [number, boolean]' is not assignable to type '[y: string]'.
-!!! error TS2322:       Type '[number, boolean]' is not assignable to type '[y: string]'.
-!!! error TS2322:         Source has 2 element(s) but target allows only 1.
+!!! error TS2322:     Type '[string] | [number]' is not assignable to type '[y: string]'.
+!!! error TS2322:       Type '[number]' is not assignable to type '[y: string]'.
+!!! error TS2322:         Type 'number' is not assignable to type 'string'.

```

Comment:
This is a change of the error for this case:
```ts
declare let f1: (x: string, ...args: [string] | [number, boolean]) => void;
declare let f2: (x: string, y: string) => void;

f1 = f2
```

With the current changes in this PR, I create a slice from the target tuple and I cap it to the source's length~. I do that for similar reasons that I've outlined in the other comment. The source signature can freely ignore the "extra" arguments so we don't even need to check them here (we'd have to ignore them through some other mechanism anyway)